internal/handlers/pg/msg_delete.go

```diff
@@ -63,10 +63,17 @@ func (h *Handler) MsgDelete(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		}
 
 		var limit int64
-		if l, _ := d.Get("limit"); l != nil {
-			if limit, err = common.GetWholeNumberParam(l); err != nil {
-				return nil, err
-			}
+
+		l, err := d.Get("limit")
+		if err != nil {
+			return nil, common.NewErrorMsg(
+				common.ErrMissingField,
+				"BSON field 'delete.deletes.limit' is missing but a required field",
+			)
+		}
+
+		if limit, err = common.GetWholeNumberParam(l); err != nil {
+			return nil, err
```

Comment:
Is it enough to just return `err`? Shouldn't it be something like `NewErrorMsg(ErrBadValue, err)`?

Let's write a test that would generate this error? I guess, it could be something like `limit` is a string or a value like `42.13`.